ios/RNSScreenStackHeaderConfig.mm

```diff
@@ -169,19 +169,17 @@ - (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event
         continue;
       }
 
-      // we wrap the headerLeft/Right component in a UIBarButtonItem
-      // so we need to hit test subviews from left to right, because of the view flattening
-      UIView *headerComponent = nil;
-      for (UIView *headerComponentSubview in subview.subviews) {
+      // We wrap the headerLeft/Right component in a UIBarButtonItem
+      // so we need to hit test subviews, because of the view flattening
+      // (we match RCTViewComponentView implementation).
+      for (UIView *headerComponentSubview in [subview.subviews reverseObjectEnumerator]) {
         CGPoint convertedPoint = [self convertPoint:point toView:headerComponentSubview];
         UIView *hitTestResult = [headerComponentSubview hitTest:convertedPoint withEvent:event];
 
         if (hitTestResult != nil) {
-          headerComponent = hitTestResult;
+          return hitTestResult;
         }
       }
```

Comment:
The comment above the for loop explicitly states, that the default behaviour of reverse iterating over the children) has been deliberately changed to left to right.

Can we do some research & see what problem exactly this approach solved & whether we do not introduce a regression with swapping this back?